documentation/CAMARA-API-access-and-user-consent.md

```diff
@@ -151,14 +151,16 @@ alt Standard OIDC Auth Code Flow between Invoker and API Exposure Platform
   alt If Consent is Granted or Consent not needed for legal basis   
     ExpO-->>FE: 302<br>Location: invoker_callback?code=Operatorcode
   else If Consent is NOT granted - Consent Capture within AuthCode Flow  
-    Note over FE,ExpO: Consent capture  
-    alt when user grants consent / otherwise flow fails  
-      ExpO-->>FE: 302<br>Location: invoker_callback?code=Operatorcode
-    end  
-  end  
+    Note over FE,ExpO: Consent capture    
```

Comment:
@eric-murray Thank you for your comment! But actually I don't see how it applies to Auth code flow. In the case of Auth code flow, the consent capture is not an asynchronous process like in, say, CIBA. In fact, in the case of Auth code flow, the application is redirected to a "consent capture page" where the user can browse and grant or revoke consent for the requested resources.
And what is particularly clarified in this PR is that if the user doesn't consent to the access (where explicit consent is required by the legal basis), the code is still provided in the GET /authorize response.

Reply:
Hi @jpengar

It was precisely because an authorisation code is returned whether or not the end user consents that I had assumed the "Consent Capture" did not need to be synchronous.

Let's see what the OpenId Connect core [specification](https://openid.net/specs/openid-connect-core-1_0.html#Consent) has to say about this:

> Once the End-User is authenticated, the Authorization Server MUST obtain an authorization decision before releasing information to the Relying Party. When permitted by the request parameters used, this MAY be done through an interactive dialogue with the End-User that makes it clear what is being consented to or by establishing consent via conditions for processing the request or other means (for example, via previous administrative consent). Sections [2](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) and [5.3](https://openid.net/specs/openid-connect-core-1_0.html#UserInfo) describe information release mechanisms.

So "an interactive dialogue" is but one option for OpenId Connect. But I now understand that CAMARA are mandating this approach, and that "Consent Capture" consequently must be synchronous?

This is an important point of difference from the OpenId Connect specifications which needs to be properly documented by CAMARA. Specifying this as "Consent Capture" is insufficient.

Reply:
@eric-murray To date, the agreed upon and defined Auth code flow captures user consent (when required) via an interactive browsing dialog to which the user is redirected (as documented).
Also, there is nothing preventing an application from executing the Auth code flow at any time, not necessarily at the same time the network resource (CAMARA API) is consumed. So, for example, consent could be obtained at the time of user onboarding.

In any case, this topic is not related to the specific change being made is this PR. "Consent Capture was also specified before this change. So please, let's not block this PR change because of this parallel discussion. Would you please open a new issue and contribute with your documentation improvement suggestion to avoid confusion?

Reply:
> I would suggest to clarify the "Consent capture" process by rewording this to "Start end user consent capture process following Section 3.1.2.4 of the OpenID Connect Core 1.0 specification".

@eric-murray I'm fine with your rewording suggestion. Added in commit https://github.com/camaraproject/IdentityAndConsentManagement/pull/86/commits/1ead9a53d2eaa70c2a415908dd64f39e63320868